File: kotlin/ktor-todo-list-api/src/main/kotlin/todolist/model/TodoService.kt
package todolist.model

class TodoService(private val repository: TodoRepository) {
    fun createTodo(request: TodoCreate): Todo {
        return repository.createTodo(request)
    }

    fun fetchTodos(): List<Todo> {
        return repository.findAllTodos()
    }

    fun fetchTodoById(id: Int): Todo? {
        return repository.findTodoById(id)
    }

    fun updateTodoById(id: Int, request: TodoUpdate): Todo? {
        return repository.updateTodoById(id, request)
    }

    fun deleteTodoById(id: Int): Boolean {
        return repository.deleteTodoById(id)
    }
}

File: typescript/express-zod-todo-list-api/src/todos/todo.service.ts
import { Todo, TodoCreate, TodoUpdate } from "./todo.model";
import * as repository from "./todo.repository";

export function createTodo(request: TodoCreate): Todo {
  return repository.createTodo(request);
}

export function fetchTodos(): Todo[] {
  return repository.findAllTodos();
}

export function fetchTodoById(id: number): Todo | undefined {
  return repository.findTodoById(id);
}

export function updateTodoById(id: number, request: TodoUpdate): Todo | undefined {
  return repository.updateTodoById(id, request);
}

export function deleteTodoById(id: number): boolean {
  return repository.deleteTodoById(id);
}
